src/huggingface_hub/hub_mixin.py

```diff
@@ -221,55 +263,82 @@ def push_to_hub(
-                class.
-            organization (`str`, *optional*):
-                Organization in which you want to push your model or tokenizer
-                (you must be a member of this organization).
+                Message to commit while pushing.
             private (`bool`, *optional*):
                 Whether the repository created should be private.
             api_endpoint (`str`, *optional*):
                 The API endpoint to use when pushing the model to the hub.
-            use_auth_token (`bool` or `str`, *optional*):
+            token (`str`, *optional*):
                 The token to use as HTTP bearer authorization for remote files.
-                If `True`, will use the token generated when running
-                `transformers-cli login` (stored in `~/.huggingface`). Will
-                default to `True` if `repo_url` is not specified.
-            git_user (`str`, *optional*):
-                will override the `git config user.name` for committing and
-                pushing files to the hub.
-            git_email (`str`, *optional*):
-                will override the `git config user.email` for committing and
-                pushing files to the hub.
+                If not set, will use the token set when logging in with
+                `transformers-cli login` (stored in `~/.huggingface`).
+            branch (`str`, *optional*):
+                The git branch on which to push the model. This defaults to
+                the default branch as specified in your repository, which
+                defaults to `"main"`.
+            create_pr (`boolean`, *optional*):
+                Whether or not to create a Pull Request from `branch` with that commit.
+                Defaults to `False`.
             config (`dict`, *optional*):
                 Configuration object to be saved alongside the model weights.
-            skip_lfs_files (`bool`, *optional*, defaults to `False`):
-                Whether to skip git-LFS files or not.
-
 
         Returns:
             The url of the commit of your model in the given repository.
         """
+        # Repo id is set means we use the new version using HTTP endpoint
+        # (introduced v0.9)
+        if repo_id is not None:
+            token, _ = hf_api._validate_or_retrieve_token(token)
+            api = HfApi(endpoint=api_endpoint)
+
+            api.create_repo(
+                repo_id=repo_id,
+                repo_type="model",
+                token=token,
+                private=private,
+                exist_ok=True,
+            )
+
+            # Push the files to the repo in a single commit
+            with tempfile.TemporaryDirectory() as tmp:
+                saved_path = Path(tmp) / repo_id
+                self.save_pretrained(saved_path, config=config)
+                return api.upload_folder(
+                    repo_id=repo_id,
+                    repo_type="model",
```

Comment:
No preference, but this is not needed. (maybe being explicit is better as you do here)